packages/gatsby/src/bootstrap/load-plugins/__tests__/load-plugins.js

```diff
@@ -71,4 +71,97 @@ describe(`Load plugins`, () => {
 
     expect(plugins).toMatchSnapshot()
   })
+
+  describe(`TypeScript support`, () => {
+    it(`loads gatsby-plugin-typescript if not provided`, async () => {
+      const config = {
+        plugins: [],
+      }
+
+      let plugins = await loadPlugins(config)
+
+      plugins = replaceFieldsThatCanVary(plugins)
+
+      expect(plugins).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            browserAPIs: [],
+            id: ``,
+            name: `gatsby-plugin-typescript`,
+            nodeAPIs: [
+              `resolvableExtensions`,
+              `onCreateBabelConfig`,
+              `onCreateWebpackConfig`,
+            ],
+            pluginOptions: {
+              plugins: [],
+            },
+            resolve: ``,
+            ssrAPIs: [],
+            version: `1.0.0`,
+          }),
+        ])
+      )
+    })
+
+    it(`uses the user provided plugin-typescript if provided`, async () => {
+      const config = {
+        plugins: [
+          {
+            resolve: `gatsby-plugin-typescript`,
+            options: {
+              jsxPragma: `h`,
+            },
+          },
+        ],
+      }
+
+      let plugins = await loadPlugins(config)
+
+      plugins = replaceFieldsThatCanVary(plugins)
+
+      expect(plugins).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            browserAPIs: [],
+            id: ``,
+            name: `gatsby-plugin-typescript`,
+            nodeAPIs: [
+              `resolvableExtensions`,
+              `onCreateBabelConfig`,
+              `onCreateWebpackConfig`,
+            ],
+            pluginOptions: {
+              plugins: [],
+              jsxPragma: `h`,
+            },
+            resolve: ``,
+            ssrAPIs: [],
+            version: `1.0.0`,
+          }),
+        ])
+      )
+    })
+
+    it(`does not add gatsby-plugin-typescript if it exists in config.plugins`, async () => {
+      const config = {
+        plugins: [
+          `gatsby-plugin-typescript`,
+          { resolve: `gatsby-plugin-typescript` },
+        ],
+      }
+
+      let plugins = await loadPlugins(config)
+
+      plugins = replaceFieldsThatCanVary(plugins)
+
+      const tsplugins = plugins.filter(
+        plugin => plugin.name === `gatsby-plugin-typescript`
+      )
+
+      // TODO: I think we should probably be de-duping, so this should be 1.
+      // But this test is mostly here to ensure we don't add an _additional_ gatsby-plugin-typescript
```

Comment:
I might need backtrack my previous comment about deduping - looking at how we do it in https://github.com/gatsbyjs/gatsby/blob/c36b3945c884fa35fed17a111a35f09a6262c6df/packages/gatsby/src/utils/merge-gatsby-config.js#L51-L55

We might already dedupe ... but the code is in different module, so it's not used in the code that runs in those tests. Most of gatsby-config merging make sense to do it where it is, but plugin deduping might be beneficial to move into `load-plugins.js`? (but not for this PR - that's separate concern)

Reply:
I think the plugins array we get here might already be deduped is what I mean :)